### PR TITLE
Pull in gregor avdl

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this brings in the gregor avdls into our flow typing. To test you can uncomment the code in the action below and change .md to .mdfoo and see it complain

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -9,6 +9,7 @@
 /shared/libs/flow-interface.js
 /shared/constants/types/keybase-v1.js
 /shared/constants/types/flow-types.js
+/shared/constants/types/flow-types-gregor.js
```
Comment (chrisnojima) [inline .eslintignore]: this stuff actually passes but is driven by other teams so leaving this ignored for now

File: protocol/bin/flow.js [hunk 2]
```diff
@@ -3,38 +3,38 @@
...
-.filter(jsonOnlySkipGregor)
-.map(load)
-.map(analyze)
-.reduce((acc, typeDefs) => acc.concat(typeDefs), [])
-.then(t => t.sort())
-.then(makeRpcUnionType)
-.then(write)
-
-let incomingMaps = {}
+var projects = [
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: trying to keep things simple here. just pulling in this other file and working on it. this is unlikely to change anytime soon

File: protocol/bin/flow.js [hunk 3]
```diff
@@ -3,38 +3,38 @@
...
+  .then(typeDefs => write(typeDefs, project))
+})
 
 function jsonOnlySkipGregor (file) {
-  return file !== 'gregor.json' && file !== "gregor_ui.json" && !!file.match(/.*\.json$/)
-}
-
-function load (file) {
-  return fs.readFileAsync(path.join(root, file)).then(JSON.parse)
+  return !!file.match(/.*\.json$/)
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: we handle gregor files so bring back the old filter

File: protocol/bin/js.js [hunk 4]
```diff
@@ -44,10 +44,14 @@ function write () {
 
 `)
     Object.keys(protocols).forEach(function (p) {
-      stream.write('export const ' + p + ' = ' + JSON.stringify(protocols[p], null, 2).replace(/\"/g, '\'') + '\n\n')
+      stream.write('export const ' + p + ' = ' + JSON.stringify(protocols[p], null, 2)
+        .replace(/"/g, '\'')
```
Comment (chrisnojima) [inline protocol/bin/js.js]: JSON doens't allow trailing commas so we inject it in

File: protocol/bin/js.js [hunk 5]
```diff
@@ -57,5 +61,5 @@ function fixCase (s) {
   function replace (s) {
     return s[1][0].toUpperCase() + s[1].substr(1)
   }
-  return s.toLowerCase().replace(/(\_\w)/g, replace)
+  return s.toLowerCase().replace(/(_\w)/g, replace)
```
Comment (chrisnojima) [inline protocol/bin/js.js]: we actually don't need the \ here

File: shared/actions/config/index.js [hunk 6]
```diff
@@ -71,6 +71,23 @@ export function bootstrap (): AsyncAction {
         [dispatch(getCurrentStatus()), dispatch(getExtendedStatus()), dispatch(getConfig())]).then(() => {
           dispatch({type: Constants.bootstrapped, payload: null})
           dispatch(navBasedOnLoginState())
+
+          // TEMP to test flow, delete when we get thumbs
```
Comment (chrisnojima) [inline shared/actions/config/index.js]: - [ ] remove this when we get thumbs

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by cjb** [comment]
:thumbsup: % remove test code

**Pull request merged**